### ISSUE TITLE
Search freezes (on Mac)

### ISSUE THREAD

**Issue by kaifastromai** [Canop/broot#680]
From what I've read. the maintainer/developer does not use a mac, but there appears to be a bug in broot where when searching, it will (some times) just freeze for an extended period of time (maybe 2-3 minutes). It usually eventually recovers, but I was under the impression that the ui should not freeze when a search is happening...
I am running broot v. 1.20.2 (via brew) on MacOS ventura 
on alacrity. And it doesn't seem related to /proc or / search bug since everything I've searched so far is just ~. 

This is probably not very actionable, but thought I should let you no.

**Comment by Canop** [comment]
The most probable reason for a freeze is when the low level function reading a directory freezes. This may happen on some special directories (we had such a problem on `/proc` on linux with some versions of rust). Can you tell if it's happen when specific directories are in the observable path (children of the current tree root) ?

**Comment by kaifastromai** [comment]
I don't believe so. It's just in my home directory, so no special paths I am aware of.

**Comment by Canop** [comment]
If it happens again, the debug log (https://dystroy.org/broot/community/#log) would show which directory blocks; on macOS, iCloud managed directories under ~/Library are a likely suspect. Closing.
